### ISSUE TITLE
HTTPS connection is rejected in my server

### ISSUE THREAD

**Issue by ghost** [misskey-dev/misskey#2791]
# Summary

I have set up my server and solved the issue https://github.com/syuilo/misskey/issues/2787, now I face another puzzle. My server rejects HTTP and HTTPS connections.

# Expected Behavior
`https://5.distsn.org:2001` shows the top page of Misskey.

# Actual Behavior
`https://5.distsn.org:2001` rejects the connection.

`http://5.distsn.org:2002` returns the empty reply.

# Steps to Reproduce
1. Set up my server.
2. Configure systemd service, enable and start it.

# Environment
```
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04.5 LTS"
```

# Log

`sudo journalctl -u misskey` log:

```
Sep 30 18:18:52 ik1-312-16250 systemd[1]: Started Misskey daemon.
Sep 30 18:18:52 ik1-312-16250 sudo[1766]:  misskey : TTY=unknown ; PWD=/home/misskey/misskey ; USER=root ; COMMAND=/usr/bin/npm start
Sep 30 18:18:52 ik1-312-16250 sudo[1766]: pam_unix(sudo:session): session opened for user root by (uid=0)
Sep 30 18:18:53 ik1-312-16250 misskey[1766]: > misskey@8.63.0 start /home/misskey/misskey
Sep 30 18:18:53 ik1-312-16250 misskey[1766]: > node ./built
Sep 30 18:18:54 ik1-312-16250 misskey[1766]: [18:18:54 INFO] Welcome to Misskey!
Sep 30 18:18:54 ik1-312-16250 misskey[1766]: [18:18:54 INFO] [Deps] Node.js v10.10.0
Sep 30 18:18:54 ik1-312-16250 misskey[1766]: [18:18:54 INFO] [Machine] Hostname: ik1-312-16250
Sep 30 18:18:54 ik1-312-16250 misskey[1766]: [18:18:54 INFO] [Machine] Platform: linux
Sep 30 18:18:54 ik1-312-16250 misskey[1766]: [18:18:54 INFO] [Machine] Architecture: x64
Sep 30 18:18:54 ik1-312-16250 misskey[1766]: [18:18:54 INFO] [Machine] CPU: 3 core
Sep 30 18:18:54 ik1-312-16250 misskey[1766]: [18:18:54 INFO] [Machine] MEM: 2.0GB (available: 1.6GB)
Sep 30 18:18:54 ik1-312-16250 misskey[1766]: [18:18:54 INFO] [Env] NODE_ENV is not set
Sep 30 18:18:54 ik1-312-16250 misskey[1766]: [18:18:54 WARN] [Env] The environment is not in production mode
Sep 30 18:18:54 ik1-312-16250 misskey[1766]: [18:18:54 WARN] [Env] Do not use for production purpose
Sep 30 18:18:54 ik1-312-16250 misskey[1766]: [18:18:54 INFO] [Env] You have root privileges
Sep 30 18:18:54 ik1-312-16250 misskey[1766]: [18:18:54 INFO] [Config] Loaded
Sep 30 18:18:54 ik1-312-16250 misskey[1766]: [18:18:54 INFO] [MongoDB] Connecting to mongodb://misskey:****@localhost:27017/misskey
Sep 30 18:18:54 ik1-312-16250 misskey[1766]: [18:18:54 INFO] Misskey initialized
Sep 30 18:18:54 ik1-312-16250 misskey[1766]: [119B blob data]
Sep 30 18:18:57 ik1-312-16250 misskey[1766]: [1.3K blob data]
Sep 30 18:18:57 ik1-312-16250 misskey[1766]: [18:18:57 INFO] All workers started
Sep 30 18:18:57 ik1-312-16250 misskey[1766]: [18:18:57 INFO] Now listening on port 2002 on https://5.distsn.org:2001
```

Log in my local PC:

```
vaginaplant@mute:~$ curl https://5.distsn.org:2001
curl: (7) Failed to connect to 5.distsn.org port 2001: 接続を拒否されました
vaginaplant@mute:~$ curl http://5.distsn.org:2002
curl: (52) Empty reply from server
```

**Comment by mei23** [comment]
May be configuration(`.config/default.yml`) mismatch.

- Listen port decided by `port` section.
- Listen protocol decided by `https` section. (define or undefined)
- `url` indicates URL for client browser. (Unrelated with listen port/protocol)
  - `url` scheme are allways https.
  - `url` not contains port.

For example ...
```
url: https://5.distsn.org/
port: 2001
# comment out
# https: 
#  :
```
=> Listen on http://*:2001/

```
url: https://5.distsn.org/
port: 2001
# define https
https:
  :
```
=> Listen on https://*:2001/

**Comment by ghost** [comment]
For recording my operations, now my `default.yml` is:
```
url: https://5.distsn.org:2001

# A port that your Misskey server should listen.
# This value is not a port to use when accessing with a browser.
port: 2002
```
(omission)
```
# TLS
https:
  # path for certification
  key: /etc/letsencrypt/live/5.distsn.org/privkey.pem
  cert: /etc/letsencrypt/live/5.distsn.org/cert.pem
```

**Comment by ghost** [comment]
And now I comment outed the `https` section, `http://5.distsn.org:2002/` shows the top page!

**Comment by ghost** [comment]
Finally my `defualt.yml` is:
```
url: https://5.distsn.org/

# A port that your Misskey server should listen.
# This value is not a port to use when accessing with a browser.
port: 443
```
(omittion)
```
# TLS
https:
  # path for certification
  key: /etc/letsencrypt/live/5.distsn.org/privkey.pem
  cert: /etc/letsencrypt/live/5.distsn.org/cert.pem
```

And now https://5.distsn.org is available! Thank you very much.

**Comment by ghost** [comment]
Is this ascii art in the `exsample.yml` is outdated, or future specification?
```
# URL and Port settings overview
# e.g., If you want to realize following structure:
#
#               +--- https://example.com:123 ----------+
# +------+      |+-------------+      +---------------+|
# | User | ---> || Proxy (123) | ---> | Misskey (456) ||
# +------+      |+-------------+      +---------------+|
#               +--------------------------------------+
#
# You need to set 'https://example.com:123' to 'url' prop and
# You need to set 456 to 'port' prop.
#
# In other words, the 'url' prop should be the final accessible URL seen by a user.
# 'port' prop is a port that the Misskey server should actually listen
# on and it is not necessarily the port that a user accesses.
```
